files/system/usr/share/ublue-os/just/60-custom.just

```diff
@@ -78,6 +78,65 @@ harden-flatpak:
         flatpak override --user --env=LD_PRELOAD=/var/run/host/usr/lib64/glibc-hwcaps/x86-64-v"$bestuarch"/libhardened_malloc.so
     fi
 
+# Check for unsafe flathub repos and flatpak permissions
+audit-flatpak:
+	#!/bin/bash
+	function hasPermission {
+		local permissions=$1
+		local prefix=$2
+		local query=$3
+		local line=$(grep "^${prefix}=" <<< "$permissions" | sed -e "s/^${prefix}=//" -e "s/#.*//")
+		IFS=';' read -r -a list <<< "$line"
+		for p in ${list[@]}; do
+			if [[ "$p" == "$query" ]]; then
+				return
+			fi
+		done
+		return 1
+	}
+	
+	echo "Checking remotes..."
+	remotes="$(flatpak remotes -d)"
+	while read -r remote ; do
+		ref="$(cut -f 1 <<<"$remote")"
+		url="$(cut -f 3 <<< "$remote")"
+		subset="$(cut -f 5 <<< "$remote")"
+			if [ "$subset" != "verified" ]; then
+				echo "Warning: $ref is not a verified repo"
+			fi
+			if [[ "$url" != "https://dl.flathub.org/repo/" && "$url" != "https://dl.flathub.org/beta-repo/" ]]; then
+				echo "Warning: $ref is configured with an unknown url"
+			fi
+	done <<< "$remotes"
+	
+	echo "Checking permissions. This may take some time..."
+	declare -A flatpaks
+	while read -r ref version; do
+		flatpaks+=(["${ref}"]="${ref}//${version}")
+	done <<<$(flatpak list | sort -k 1 | cut --fields 2,4)
+	declare -a network_flatpaks
+	declare -a x11_flatpaks
+	for f in ${!flatpaks[@]}; do
+		fullref=${flatpaks["$f"]}
+		permissions=$(flatpak info --show-permissions "$fullref")
+		if hasPermission "$permissions" "shared" "network"; then
+			network_flatpaks+=($f)
+		fi
+		if hasPermission "$permissions" "sockets" "x11" && ! hasPermission "$permissions" "sockets" "fallback-x11"  ]]; then
+			x11_flatpaks+=($f)
+		fi
+	done
+	echo "The following flatpaks have network access:"
+	for flatpak in "${network_flatpaks[@]}"; do
+		echo -e "\t${flatpak}"
+	done
+	echo "The following flatpaks have x11 access:"
```

Comment:
looks good, one thing though is that if none of your flatpaks have one of the permissions, it shows this which is weird:

```
The following flatpaks have x11 access:
Completed audit.
```

I think it should say "No flatpaks have x11 access"


Alternatively, we could do it per-flatpak which would make the UI similar to audit-secureblue?


Something like this?

```
Auditing org.gnome.Boxes...                          [ WARNING ]
org.gnome.Boxes has network access!
Auditing org.gnome.Evince...                          [ SUCCESS ]
```

what do you think?